### ISSUE TITLE
Implement sync pragmas to keybindings

### ISSUE THREAD

**Issue by bernardo-amaral** [shanalikhan/code-settings-sync#800]
💡 **Is your feature request related to a problem? Please describe.**
In the Settings, the user can choice what he wants sync, with false and true values.

💡 **Describe the solution you'd like**
- If the user wants not sync settings, that data will not will be uploaded to your gist.
- If the user wants not sync extensions, that data will not will be uploaded to your gist.
- If the user wants not sync key binds, that data will not will be uploaded to your gist.

💡 **Additional context**
Nothing additional!

**Comment by njkevlani** [comment]
This can be done by adding a setting like `sync.excludeSettings` which is a list of settings that are not supposed to be synced.
For example, to exclude font setting,

```json
{
    "sync.excludeSettings": [
        "editor.fontSize"
    ]
}
```

Something similar can be implemented for keybindings and extensions as well.

**Comment by shanalikhan** [comment]
**Custom Sync for `settings.json`**

If you are looking to exclude the settings keys from being sync.  See the details here :

https://github.com/shanalikhan/code-settings-sync/wiki/Sync-Pragmas#ignore-settings

**Custom Sync for Extensions**

If you want to disable the extension sync, set `sync.syncExtensions` to `false` to disable the extensions.

If you want to remove some extensions from sync process. Add your extension complete name in `ignoreExtensions` ( Settings Sync Global Settings - check readme )

**Custom Sync for keybindings**

I'm open for PR. This feature needs to be added.

**Comment by shanalikhan** [comment]
For keybindings, i would like you to be-specific with your scenarios and example about what you are looking to achieve.

**Comment by njkevlani** [comment]
I think it would be better if we use `@sync-ignore` like implementation for `keybinding.json`.

Something like https://gist.github.com/njkevlani/d79973d50258fe4312786b198746da3f#file-keybindings-json

**Comment by MstrBlonde** [comment]
> If you want to remove some extensions from sync process. Add your extension complete name in `ignoreExtensions` ( Settings Sync Global Settings - check readme )

I can't figure out what "complete name" is for any given extension, can you please clarify?

**Comment by njkevlani** [comment]
@MstrBlonde 

"complete name" = Unique Identifier without publisher id.

for example see https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync

Unique Identifier: Shan.code-settings-sync

So the "complete name" = `code-settings-sync`

**Comment by shanalikhan** [comment]
Yes this can be done for `keybindings.json`.

P.S : Public edit is enabled for Wiki. If you guys think documentation can be improved, just edit those pages and improve it, that would be great help to other users :)

**Comment by njkevlani** [comment]
@shanalikhan 

Functions `processBeforeUpload` and `processBeforeWrite` are used to add functionality **Sync Pragmas** in `settings.json`.

I think we can use `processBeforeUpload` and `processBeforeWrite` for this purpose as well. Or do I have to rewrite similar functions with some changes? 

If the same functions can be used for both `settings.json` and `keybindings.json`, I'll just change the variable name where ever appropriate and use them for `keybindings.json` as well.

LMK.

**Comment by shanalikhan** [comment]
No these function should work for all of the content string that will be passed.
You just need understand how these functions are being called in which location for `settings.json` and implment the calling layout for `keybindings.json`

Make sure you write the test cases for it.

If you are working on to implement this issue, it would be great if you can look into other ticket as well regarding `keybindings.json` - https://github.com/shanalikhan/code-settings-sync/issues/515.

**Comment by njkevlani** [comment]
Will open a PR by this weekend :)

**Comment by njkevlani** [comment]
How do I run test? [This Method](https://code.visualstudio.com/api/working-with-extensions/testing-extension) for test fails on master branch

**Comment by shanalikhan** [comment]
@njkevlani can you fix fail test method also, i think some PR broke it.

**Comment by shanalikhan** [comment]
I have released new version of Settings Sync v3.3.0 fixing this issue. Im closing this ticket let me know if u guys run into some problem.

Anyone can improve public wiki to help community
